#38111 PyChop MVP Refactoring

lr154lrose opened the pull request:
### Description of work
Refactoring PyChop to conform to the Model View Presenter design

#### Summary of work
<!-- Please provide a short, high level description of the work that was done.
-->
UI logic has been separated from business logic, and a presenter has been created to help the communication between the two. Methods that are purely containing UI logic (View) have been left in PyChopGui.py, but everything that was used to communicate to the "Model" and the scripts, has been moved to the "Presenter". 
Functions where calculations are required go through the Presenter which will send back the result to the View. Direct communication between PyChopGui.py and the scripts where calculations take place has been moved to Presenter.


#### Purpose of work
See github issue #38111 


<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has been undertaken.
-->

### To test:

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

Commit:
Pychop refactoring code cleand up

Commit:
changes to the self.engine object and added comments for future reference when it comes to helper functions

RichardWaiteSTFC reviewed (changes requested):
Thanks for all your hard work on this - definitely big improvements to separate out business logic from the view - which looking at the code for the first time was no easy task!

I know this is a work in progress but I'm treating this PR as a way of keeping track of decisions made and next steps. I don't expect you to act on any of these comments/suggestions!

1. I think the `setInstrument` method in the view is quite large and perhaps part of it could be split into smaller methods (e.g. `set_frequencies`, `set_phases` - I apologise if those names are nonsense!)
2. In fact there appears to be some logic in `setInstrument` (e.g. quite a lot of if/else which depends on certain fields in the `data` dict returned from the presenter, and some HYSPEC specific code). I wonder if having split the method into smaller methods in (1) whether these should be called from the presenter (ultimately this logic would live in the model/instrument and be passed to the view via the presenter)?
3. The same could also possibly be said of the MERLIN specific stuff in the `setFreq` method in the view.

Other than that there are a few commented out sections off code that need to be clarified (should they be removed, or is that code that is required but currently broken).

Commit:
removed commented out code that was moved to Presenter
- also removed MVP example function that came from the MVP tool that creates example files
- changed copyright year to 2025

SilkeSchomann commented:
This PR cannot be closed right now.

sf1919 commented:
To be sorted

SilkeSchomann commented:
We still want to keep this.
